liblangutil/SourceReferenceExtractor.cpp

```diff
@@ -44,16 +45,15 @@ SourceReferenceExtractor::Message SourceReferenceExtractor::extract(
 		for (auto const& info: secondaryLocation->infos)
 			secondary.emplace_back(extract(_charStreamProvider, &info.second, info.first));
 
-	return Message{std::move(primary), _severity, std::move(secondary), nullopt};
+	return Message{std::move(primary), _typeOrSeverity, std::move(secondary), nullopt};
 }
 
 SourceReferenceExtractor::Message SourceReferenceExtractor::extract(
 	CharStreamProvider const& _charStreamProvider,
 	Error const& _error
 )
 {
-	string severity = Error::formatErrorSeverity(Error::errorSeverity(_error.type()));
-	Message message = extract(_charStreamProvider, _error, severity);
+	Message message = extract(_charStreamProvider, util::Exception(_error), _error.type());
```

Comment:
```suggestion
	Message message = extract(_charStreamProvider, static_cast<util::Exception>(_error), _error.type());
```

Reply:
I see you now reverted back to duplicating the code. Why? I just checked to make sure it compiles and it does:
```c++
	string severity = Error::formatErrorSeverity(Error::errorSeverity(_error.type()));
	Message message = extract(_charStreamProvider, static_cast<util::Exception>(_error), _typeOrSeverity);
	message.errorId = _error.errorId();
	return message;
```